actions/container-build_test_release/docker-action/build_test.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/sh
+# Copyright 2023 NWChemEx-Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Wraps the process for configuring, building, and testing an NWX repo
+#
+# Usage:
+#   build_test.sh
+#
+# Other variables:
+#   cmake_version: the version of cmake being used in the format x.y.z
+#
+
+set -e # Exit with error if any command fails
+
+# checkout repo with the current branch (not master!)
+
+git clone -b ${env_branch_name} https://github.com/NWChemEx-Project/ParallelZone.git
+cd ParallelZone
+
+cmake_command=cmake
+ctest_command=ctest
+toolchain_file=$(pwd)/toolchain.cmake
+
+
+echo "set(BUILD_TESTING ON)" > "${toolchain_file}"
+{
+  echo "set(CMAKE_CXX_STANDARD 17)"
+  echo 'set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)'
+  echo "set(BUILD_SHARED_LIBS ON)"
+  echo "set(CATCH_ENABLE_COVERAGE ON)"
+  echo "set(CMAKE_PREFIX_PATH /install)"
+  echo 'set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --coverage -std=c++17")'
+  echo 'set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DOMPI_SKIP_MPICXX")'
+  echo 'set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} --coverage")'
+  echo 'set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fprofile-arcs")'
+  echo 'set(CMAIZE_GITHUB_TOKEN '${env_cmaize_github_token}')'
+  echo 'set(CMAKE_BUILD_TYPE Release)'
```

Comment:
Just wanted to check. Is the plan to have the containers for only release version?

Reply:
I'm not sure whether the debug build mode is useful or not in CI. I think the debug mode is for development only, right? I could be wrong.

Reply:
I have seen [arguments](https://softwareengineering.stackexchange.com/a/142852) suggesting having both Release and Debug to be built with CI, but I don't think it is that important.

Reply:
For now I think it's fine to just do release, but like @keceli said there's reasons to do both eventually (the simplest of which is the resulting binary is literally different and can in rare cases have bugs in one mode or the other).